functions/Set-DbaLogin.ps1

```diff
@@ -179,8 +179,10 @@ function Set-DbaLogin {
                 "PSCredential" { $newPassword = $Password.Password }
                 "SecureString" { $newPassword = $Password }
             }
-        }
-        else {
+
+            if ($Password.GetType().Name -notin @('PSCredential', 'SecureString')) {
+                Stop-Function -Message "Password must be a PSCredential or SecureString" -Target $Login
```

Comment:
I would just throw the Stop-Function call into the switch statement, it would work either way. Just seems backward for us to go through and validate the type name twice.

Something like this:
```
        if (Test-Bound -ParameterName Password) {
            switch ($Password.GetType().Name) {
                "PSCredential" { $newPassword = $Password.Password }
                "SecureString" { $newPassword = $Password }
                default {
                    Stop-Function -Message "Password must be a PSCredential or SecureString object" -Target $Login
                }
            }
        }
```

Reply:
Just a bit easier to read to me.

Reply:
Makes sense, will do. Thanks!